authentication/authentication.py

```diff
@@ -0,0 +1,22 @@
+from django.utils.timezone import now
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.exceptions import AuthenticationFailed
+from rest_framework.authtoken.models import Token
+
+from forum import settings
+
+
+class ExpiringTokenAuthentication(TokenAuthentication):
+    def authenticate_credentials(self, key):
+        try:
+            token = Token.objects.get(key=key)
+        except Token.DoesNotExist:
+            raise AuthenticationFailed("Invalid token")
+        
+        if not token.user.is_active:
```

Comment:
Is it possible situation when user can have token without activation event?

I hope no, so this condition is not needed )

Reply:
This code is originally from authentication.py of rest framework. And I didn't pay attention to this verification, only changed the error message. But user.is_active is validated in token serializer. I've removed this verification. Thank you.